internal/bminventory/inventory.go

```diff
@@ -1362,6 +1362,76 @@ func (b *bareMetalInventory) updateFreeAddressesReport(ctx context.Context, host
 	return nil
 }
 
+func vipMismatchError(apiVip, ingressVip string, cluster *common.Cluster) error {
+	return errors.Errorf("Got VIPs different than those that are stored in the DB for cluster %s. APIVip = %s @db = %s, IngressVIP = %s @db = %s",
+		cluster.ID.String(), apiVip, cluster.APIVip, ingressVip, cluster.IngressVip)
+}
+
+func (b *bareMetalInventory) processDhcpAllocationResponse(ctx context.Context, host *models.Host, dhcpAllocationResponseStr string) error {
+	var (
+		err                   error
+		dhcpAllocationReponse models.DhcpAllocationResponse
+		cluster               common.Cluster
+	)
+	log := logutil.FromContext(ctx, b.log)
+	if err = b.db.Take(&cluster, "id = ?", host.ClusterID.String()).Error; err != nil {
+		log.WithError(err).Warnf("Get cluster %s", host.ClusterID.String())
+		return err
+	}
+	if !swag.BoolValue(cluster.VipDhcpAllocation) {
+		log.Warnf("DHCP not enabled in cluster %s", host.ClusterID.String())
+		return nil
+	}
+	if err = json.Unmarshal([]byte(dhcpAllocationResponseStr), &dhcpAllocationReponse); err != nil {
+		log.WithError(err).Warnf("Json unmarshal dhcp allocation from host %s", host.ID.String())
+		return err
+	}
+	apiVip := dhcpAllocationReponse.APIVipAddress.String()
+	ingressVip := dhcpAllocationReponse.IngressVipAddress.String()
+	isApiVipInMachineCIDR, err := network.IpInCidr(apiVip, cluster.MachineNetworkCidr)
+	if err != nil {
+		log.WithError(err).Warn("Ip in CIDR for API VIP")
+		return err
+	}
+
+	isIngressVipInMachineCIDR, err := network.IpInCidr(ingressVip, cluster.MachineNetworkCidr)
+	if err != nil {
+		log.WithError(err).Warn("Ip in CIDR for Ingress VIP")
+		return err
+	}
+
+	if !(isApiVipInMachineCIDR && isIngressVipInMachineCIDR) {
+		err = errors.Errorf("At least of the IPs (%s, %s) is not in machine CIDR %s", apiVip, ingressVip, cluster.MachineNetworkCidr)
+		log.WithError(err).Warn("IP in CIDR")
+		return err
+	}
+	switch swag.StringValue(cluster.Status) {
+	case models.ClusterStatusInsufficient, models.ClusterStatusReady:
```

Comment:
why does inventory need to know about cluster statuses? why isn't it a cluster API?

Reply:
OK. Will move